model/src/test/assets/expected-scan-results.yml

```diff
@@ -65,17 +64,28 @@ results:
     start_time: "1970-01-03T00:01:00Z"
     end_time: "1970-01-03T00:02:00Z"
     file_count: 2
-    license_findings:
+    licenses:
     - license: "license 2.1"
-      locations: []
-      copyrights:
-      - statement: "copyright 3"
-        locations: []
+      location:
+        path: ""
+        start_line: 1
```

Comment:
That's interesting, how can we have start / end lines without a path?

Reply:
This question is out of scope of this commit imo as it is supposed to be a minimal transformation and the actual location is not relevant to the test.

Reply:
It's not out of scope because your new model doesn't allow something that was possible before: A license name without *any* information about where it was found. I'm not saying it was useful to have that "feature", but we need to think about how to map that to your new model. And using an empty path to do that is a bad idea, IMO, as it does not make sense to have line numbers without a path. Likewise, using "magic" line numbers like `-1` is bad, at least in tests, where you could come up with fake / dummy numbers instead.

Reply:
> It's not out of scope because your new model doesn't allow something that was possible before:

I think the fact that something was allowed in the past is not a strong argument for requiring it to allow it in the future.

> A license name without _any_ information about where it was found. I'm not saying it was useful to have that "feature", but we need to think about how to map that to your new model. 

Why do you think we need to think about that?

> And using an empty path to do that is a bad idea, IMO, as it does not make sense to have line numbers without a path. 

I was not saying that it makes sense in general. I'm saying that it makes sense in this commit.
Can you live with that?

> Likewise, using "magic" line numbers like `-1` is bad, at least in tests, where you could come up with fake / dummy numbers instead.

That's a different thing, fully agree.

...What do you suggest?

Reply:
> I think the fact that something was allowed in the past is not a strong argument for requiring it to be allowed it in the future.

Sign 😞 Yes, that's exactly why I wrote 'I'm not saying it was useful to have that "feature"'.

> Why do you think we need to think about that?

Because this thinking could also come to the conclusion that we don't need this feature anymore. But apparently currently we do, because not all supported scanners (or our wrapper implementations) seem to support getting the text locations. So to me it seems we currently need a sane way to express "no (known) text location".

> Can you live with that?

To be frank, no.

> ...What do you suggest?

As written several times already, for tests simply come up with a fake / dummy path and line numbers.

Reply:
I did not realize this issue when we designed the new classes. I see these options to deal with scanner that do not provide the file and line numbers of license findings:
* Make the fields nullable.
* Define values that express "no data available", like "[unknown]" for the path and -1 for the line numbers.

In both cases code working with the findings class needs to handle this and I think the nullable approach would be more fail proof.

Since this is already a big commit I would be fine with fixing this separately.

Or we drop support for those scanners, because we don't really use them anyway, and some upcoming features don't work without the data, but that's maybe too bold.

Reply:
I'd vote for nullable TextLocations and against dropping scanners for now. For cases where we cannot "fake" something meaningful, like mentioned scanners that do not provide a text location, I'm fine with using `TextLocation("", -1, -1)` in this PR until we have the nullable solution. In other places like tests, we should not use `TextLocation("", -1, -1)` but fake data that could be real.

Reply:
alright, done.